crates/subspace-farmer/src/single_disk_plot.rs

```diff
@@ -0,0 +1,974 @@
+                                            "Witness must have correct size unless unless \
+                                            implementation is broken in a big way; qed",
+                                        ),
+                                    ) {
+                                        Ok(piece_witness) => piece_witness,
+                                        Err(error) => {
+                                            let piece_index = sector_id.derive_piece_index(
+                                                audit_piece_offset / PIECE_SIZE as u64,
+                                                sector_metadata.total_pieces,
+                                            );
+                                            error!(
+                                                ?error,
+                                                %piece_index,
+                                                "Failed to decode witness for piece, likely caused \
+                                                by on-disk data corruption"
+                                            );
+                                            continue;
+                                        }
+                                    };
+                                    // Decode piece
+                                    piece
+                                        .as_mut()
+                                        .view_bits_mut::<Lsb0>()
+                                        .chunks_mut(space_l.get() as usize)
+                                        .enumerate()
+                                        .par_bridge()
+                                        .for_each(|(chunk_index, bits)| {
+                                            // Derive one-time pad
+                                            let mut otp = derive_chunk_otp(
+                                                &sector_id,
+                                                &piece_witness,
+                                                chunk_index as u32,
+                                            );
+                                            // XOR chunk bit by bit with one-time pad
+                                            bits.iter_mut()
+                                                .zip(otp.view_bits_mut::<Lsb0>().iter())
+                                                .for_each(|(mut a, b)| {
+                                                    *a ^= *b;
+                                                });
+                                        });
+                                    // let solution = Solution {
+                                    // };
+                                    // solutions.push(solution);
+                                }
+                            }
+
+                            drop(solutions);
+                            // client
```

Comment:
do we need these commented code?

Reply:
Not strictly speaking, this is just a draft where certain things will be and is replaced in the next PR with uncommented code that is either the same of very similar.